lib/collection/src/hash_ring.rs

```diff
@@ -75,6 +75,16 @@ impl<T: Hash + Copy + PartialEq> HashRing<T> {
         new.add(shard);
     }
 
+    pub fn commit(&mut self) -> bool {
+        let Self::Resharding { new, .. } = self else {
+            log::warn!("committing resharding hashring, but hashring is not in resharding mode");
+            return false;
```

Comment:
Shall we return an error here?

Reply:
IMO, it's fine as-is. We have enough precondition checks already, so I wanted to make `HashRing` "infallible".